python/src/numerous/collection/numerous_document.py

```diff
@@ -0,0 +1,176 @@
+"""Class for working with numerous documents."""
+
+from typing import Any, Dict, Optional
+
+from numerous._client import Client
+from numerous.generated.graphql.fragments import CollectionDocumentReference
+from numerous.generated.graphql.input_types import TagInput
+from numerous.utils import base64_to_dict, dict_to_base64
+
+
+class NumerousDocument:
+    """
+    Represents a document in a Numerous collection.
+
+    Attributes
+    ----------
+        key (str): The key of the document.
+        collection_id (str): The id of collection document belongs to.
+        data (Optional[Dict[str, Any]]): The data of the document.
+        id (Optional[str]): The unique identifier of the document.
+        cleint (Client): The client to connect.
+        tags (Dict[str, str]): The tags associated with the document.
+
+    """
+
+    def __init__(
+        self,
+        client: Client,
+        key: str,
+        collection_id: str,
+        numerous_doc_ref: Optional[CollectionDocumentReference] = None,
+    ) -> None:
+        self.key: str = key
+        self.collection_id: str = collection_id
+        self._client: Client = client
+        self.document_id: Optional[str] = None
+
+        if numerous_doc_ref is not None:
+            dict_of_tags = {tag.key: tag.value for tag in numerous_doc_ref.tags}
+            self.data: Optional[Dict[str, Any]] = base64_to_dict(numerous_doc_ref.data)
+            self.document_id = numerous_doc_ref.id
+            self.tags: Dict[str, str] = dict_of_tags if dict_of_tags is not None else {}
```

Comment:
Do not use `typing.Dict`, instead just use `dict`. This also applies to `typing.Tuple` and `typing.List` if they are used anywhere :smile: